pkg/runner/logger.go

```diff
@@ -64,6 +64,27 @@ func WithJobLogger(ctx context.Context, jobName string, secrets map[string]strin
 	return common.WithLogger(ctx, rtn)
 }
 
+func withStepLogger(ctx context.Context, stepName string) context.Context {
+	mux.Lock()
+	defer mux.Unlock()
+
+	stepLogger := logrus.New()
+
+	parentLogger := common.Logger(ctx)
+
+	if logger, ok := parentLogger.(*logrus.Logger); ok {
+		stepLogger.SetFormatter(logger.Formatter)
+		stepLogger.SetOutput(logger.Out)
+		stepLogger.SetLevel(logger.GetLevel())
+	} else {
+		stepLogger.SetOutput(os.Stdout)
+		stepLogger.SetLevel(logrus.GetLevel())
+	}
+
+	rtn := stepLogger.WithFields(logrus.Fields{"step": stepName})
```

Comment:
> This does not change the output for our users

I'm afraid, this change forces output to stdout even if my usecase for nektos/act wants redirection.
During debugging I always branched into the second case.

The following change does work for me without side effects
```suggestion
	parentLogger := common.Logger(ctx)
	rtn := parentLogger.WithFields(logrus.Fields{"step": stepName})
```

Reply:
That would add the step to the parent logger which is logically wrong. When a step is done, no logger should have this field anymore. That's why I created a new logger as a child.

Reply:
Can you please be a bit more specific on your test case? The default logrus logger outputs to stdout, if there is no logger in the context, that should be fine.

Reply:
> The default logrus logger outputs to stdout

No I changed it in my code see https://github.com/nektos/act/pull/1015

> That would add the step to the parent logger which is logically wrong

I need to check that later.

Reply:
> That would add the step to the parent logger which is logically wrong

**No it doesn't** parentLogger.WithFields doesn't alter parentLogger.
Also common.WithLogger also doesn't alter the previous context.

Why do you think it does alter  the parent?

Reply:
I think my misunderstanding was that changing the logger is mutating, but after reading the logrus source I noticed it isn't.
So I don't have to change the logger at all, but just at fields and put that enrty into the context.
I'll update that and it should work for you as well, right?

Reply:
> I'll update that and it should work for you as well, right?

Yes

Reply:
Updated it, I hope it does work now.